spec/unit/helpers.rb

```diff
@@ -6,3 +6,11 @@ def props(providers)
     providers.map { |p| props(p) }
   end
 end
+
+def res_hash(resources)
+  Hash[resources.map { |r| [r.name, r] }]
+end
+
+def res_providers(resources)
+  resources.map { |r| r.provider && r.provider.name }
```

Comment:
`&&` is a logical AND operator. (Ruby also has `and`, which I think differs only in precedence.) `foo && bar` means "if `foo` is falsey (`false` or `nil`, basically) return `foo`, otherwise return `bar`". (Python's logical operators have very similar behaviour.) I'm using it here to avoid explicit checks for `nil` because those turn a one-liner into three lines of variable juggling.